exonum-java-binding/core/src/test/java/com/exonum/binding/runtime/ServiceIdTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.runtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.google.common.testing.NullPointerTester;
+import com.google.common.testing.NullPointerTester.Visibility;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class ServiceIdTest {
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+      "com.acme:foo-service:0.1.0",
+      "com.acme:foo-service:1.1.1-beta1",
+      "::",
```

Comment:
is it a valid service id? If yes, I would describe this special case.
Also, how about?
```
"foo:bar:"
"::0.1"
"foo::"
```

Reply:
Hm, that's a good question. I decided to allow empty coordinates in case they are not know/specified. Shall we forbid them?

Reply:
Clarified that in the docs. If they must be forbidden for some reason — let me know.